README.md

```diff
@@ -368,6 +368,40 @@ dfa.cardinality()
 len(dfa)
 ```
 
+#### DFA.contains_substring(cls, input_symbols, substring)
+
+Directly computes the minimal DFA recognizing strings containing the
+given substring.
+
+```python
+dfa = DFA.contains_substring({'a', 'n', 'o', 'b'}, 'nano')
+```
+
+#### DFA.contains_subsequence(cls, input_symbols, subsequence)
+
+Creates a DFA which accepts all words which contain a specific subsequence of symbols
+
+```python
+dfa = DFA.contains_subsequence({'a', 'b', 'c', 'd'}, 'dcba')
+```
+
+#### DFA.of_length(cls, input_symbols, min_length=0, max_length=float('inf'))
+
+Creates a DFA which accepts all words whose length is between `min_length` and `max_length`, inclusive.
+To allow infinitely long words the value `float('inf')` can be passed in for `max_length`.
+
+```python
+dfa = DFA.of_length({'0', '1'}, 4, float('inf'))
+```
+
+#### DFA.nth_from_end(cls, input_symbols, symbol, n)
+
+Creates a DFA which accepts all words whose `n`-th character from the end is `symbol`, where `n` is a positive integer.
```

Comment:
@Tagl Please add a note to the documentation here about the `ValueError` that's raised if the `symbol` does not exist within `input_symbols`.

Reply:
Isn't there a specialized `InvaildSymbol` exception somewhere?

Reply:
@eliotwrobson @Tagl Ah yes, `InvalidSymbolError`. Good call.

Reply:
Updated